[RUM-265] Browser SDK: add `storeContextsAcrossPages` option (#19731)
DataDog/documentation: content/en/real_user_monitoring/browser/modifying_data_and_context.md

{{% /tab %}}
{{< /tabs >}}

> ## Contexts life cycle

Reviewer: same wording suggestions as on the other page

> By default, global context and user context are stored in the current page memory so:

Reviewer 2: ```suggestion
By default, global context and user context are stored in the current page memory, which means they are not:
```

> - they are not kept after a full reload of the page

Reviewer 2: ```suggestion
- kept after a full reload of the page
```

> - they are not shared across different tabs / windows of the same session

Reviewer 2: ```suggestion
- shared across different tabs or windows of the same session
```

> So in order to be added to all events of the session, those contexts need to be attached on every page.

Reviewer 2: ```suggestion
To add them to all events of the session, those contexts must be attached to every page.
```

> With the introduction of the `storeContextsAcrossPages` configuration option in the v4.49.0 of the browser SDK, those contexts can be stored in local memory, allowing the following behaviours:

Reviewer 2: ```suggestion
With the introduction of the `storeContextsAcrossPages` configuration option in v4.49.0 of the browser SDK, those contexts can be stored in local memory, allowing the following behaviors:
```

- Contexts are preserved after a full reload
- Contexts are synchronized between tabs opened on the same origin

However, this feature comes with some **limitations**:

> - Setting PII in those contexts is not recommended, data stored in local storage outlives the user navigation

Reviewer 2: ```suggestion
- Setting Personal Identifiable Information (PII) in those contexts is not recommended, as data stored in local storage outlives the user navigation
```

> - Incompatibility with the `trackSessionAcrossSubdomains` options, the contexts are only shared among the same origin (login.site.com ≠ app.site.com)

Reviewer 2: ```suggestion
- The feature is incompatible with the `trackSessionAcrossSubdomains` options because the contexts are only shared among the same origin (login.site.com ≠ app.site.com)
```

> - Subject to the local storage 5 MiB limit for the whole application, so application specific data, datadog contexts and other third parties data stored in local storage need to stay under this limit to avoid any issue.

Reviewer 2: ```suggestion
- The feature is subject to the local storage 5 MiB limit for the entire application, so the application-specific data, Datadog contexts, and other third-party data stored in local storage must be within this limit to avoid any issues.
```
